tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -1491,50 +1491,72 @@
+    </Resource>
     <Resource Name="RTarget" Dimension="TEXTURE2D" Width="1280" Height="2400" Format="R32G32B32A32_FLOAT" Flags="ALLOW_RENDER_TARGET" InitialResourceState="COPY_DEST" ReadBack="true" />
+    <Resource Name="g_result"      Dimension="BUFFER"     Width="64"            InitialResourceState="COPY_DEST" Init="ZERO"      Flags="ALLOW_UNORDERED_ACCESS" ReadBack="true"  TransitionTo="UNORDERED_ACCESS"/>
+
     <DescriptorHeap Name="RtvHeap" NumDescriptors="1" Type="RTV">
       <Descriptor Name="RTarget" Kind="RTV"/>
     </DescriptorHeap>
+    <DescriptorHeap Name="ResourceDescriptorHeap" Type="CBV_SRV_UAV">
+      <Descriptor Name="g_result"   NumElements="3" Kind="UAV"  StructureByteStride="4"/>
+    </DescriptorHeap>  
     <InputElements>
       <InputElement SemanticName="POSITION" Format="R32G32B32_FLOAT" AlignedByteOffset="0" />
       <InputElement SemanticName="COLOR" Format="R32G32B32A32_FLOAT" AlignedByteOffset="12" />
     </InputElements>
     <RenderTargets>
       <RenderTarget Name="RTarget" />
     </RenderTargets>
-    <Shader Name="VS" Target="vs_6_1">
+
+    <RootValues>
+      <RootValue HeapName="ResourceDescriptorHeap" Index="0" />
+    </RootValues>
+    
+    <Shader Name="PS61" Target="ps_6_1" EntryPoint="PSMain" Text="@MAIN"/>
+    <Shader Name="VS61" Target="vs_6_1" EntryPoint="VSMain" Text="@MAIN"/>
+
+
+    <Shader Name="MAIN" Target="vs_6_1" EntryPoint="VSMain">
       <![CDATA[
       struct PSInput {
         float4 position : SV_POSITION;
         nointerpolation float4 color : COLOR;
+        uint svid : SVertexID;
       };
-      PSInput main(float4 position : POSITION, float4 color : COLOR) {
+      
+      
+      RWStructuredBuffer <uint> g_result   : register(u0);
+      
+      PSInput VSMain(float4 position : POSITION, float4 color : COLOR, uint svid : SV_VertexID) {
         PSInput result;
         result.position = position;
         result.color = color;
+        result.svid = svid;
+         
         return result;
       }
-      ]]>
-    </Shader>
-    <Shader Name="PS" Target="ps_6_1">
-      <![CDATA[
-      struct PSInput {
-        float4 position : SV_POSITION;
-        nointerpolation float4 color : COLOR;
-      };
-      float4 main(PSInput input, float3 bary : SV_Barycentrics) : SV_Target {
+    
+      float4 PSMain(PSInput input, float3 bary : SV_Barycentrics) : SV_Target {
+        float4 pos = input.position;
+        // check to make sure that the pixel shader will see the barycentric weight associated
+        // with the first vertex in the x component of the SV_Barycentric vector, and likewise for subsequent vertices
+        float4 color = input.color;
+     
+        g_result[0] = GetAttributeAtVertex(input.svid, 0);
+        g_result[1] = GetAttributeAtVertex(input.svid, 1);
+        g_result[2] = GetAttributeAtVertex(input.svid, 2);
```

Comment:
I think it's important for each pixel to get the right values here.  Since all pixels are writing to the same locations, we have a data race.  While we expect the values to be the same, that's something we should probably be testing, not assuming.

Some approaches to solve this:
- use AppendStructuredBuffer<uint3> and check that all resulting values are the same expected result.  Drawback: this might require additional setup the framework doesn't support easily.
- output results to render target and check those.  Since this is already outputting the interpolated color using the barycentrics, I think it would be best to just split this into a new shader if this approach is chosen, making it easier to test both things independently.